mmv1/third_party/terraform/tests/resource_workstations_workstation_cluster_test.go.erb

```diff
@@ -0,0 +1,78 @@
+<% autogen_exception -%>
+package google
+<% unless version == 'ga' -%>
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccorkstationsWorkstationCluster_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProvidersOiCS,
+		CheckDestroy: testAccCheckWorkstationsWorkstationClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWorkstationsWorkstationCluster_workstationClusterBasicExample(context),
+			},
+			{
+				ResourceName:            "google_workstations_workstation_cluster.default",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"etag"},
+			},
+			{
+				Config: testAccorkstationsWorkstationCluster_update(context),
+			},
+			{
+				ResourceName:            "google_workstations_workstation_cluster.default",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"etag"},
+			},
+		},
+	})
+}
+
+func testAccorkstationsWorkstationCluster_update(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_workstations_workstation_cluster" "default" {
+  provider   			 = google-beta
+  workstation_cluster_id = "workstation-cluster-%{random_suffix}"
+  network                = google_compute_network.default.id
+  subnetwork             = google_compute_subnetwork.default.id
+  location   		     = "us-central1"
+
+  labels = {
+	update = "true" 
+  }
+}
+
+resource "google_compute_network" "default" {
+  provider                = google-beta
+  name                    = "network-%{random_suffix}"
+  auto_create_subnetworks = false
+}
+
+data "google_project" "project" {
+  provider = google-beta
+}
+
+resource "google_compute_subnetwork" "default" {
+  provider      = google-beta
+  name          = "subnetwork-%{random_suffix}"
+  ip_cidr_range = "10.0.0.0/24"
+  region        = "us-central1"
+  network       = google_compute_network.default.id
+}
```

Comment:
The name of the subnetwork isn't matching up with generated name of the subnetwork in `testAccWorkstationsWorkstationCluster_workstationClusterBasicExample`... This is forcing recreation here causing the `google_workstations_workstation_cluster` to subsequently get recreated. Lets either align the names or encapsulate both configs (before and after) within this file explicitly. 

[modular-magician/terraform-provider-google-beta@`auto-pr-7005-old..auto-pr-7005`#diff-38fbf90791](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7005-old..auto-pr-7005#diff-38fbf907918e59c24a64079871d90f13c5f1d95b93924b38b483dd2dc48fc902R81)

Reply:
Sharp.. thanks Scott